scripts/test.sh

```diff
@@ -1,35 +1,185 @@
+    -ni, --no-integration-tests     Do not run integration tests.
+    -nc, --no-coverage              Do not measure test coverage.
+    -h, --help                      Show usage.
+
+    All Google Test flags are also accepted.  A subset is listed below.  For a
+    complete list of Google Test flags and their usage, run:
+    $ ./test.sh -ni -nc --gtest_help
+
+    --gtest_filter=<filter>         Define filter that specifies which tests
+                                    to run.
+                                    Default:  no filter (i.e. run all tests)
+    --gtest_repeat=<integer>        Repeat tests.
+                                    Default:  1 (i.e. do not repeat tests)
+    --gtest_break_on_failure        Stop running tests if a test fails.
+
+EXAMPLES:
+    - Run unit tests and integration tests and measure test coverage.
+      The build directory is set by an environment variable called 'BUILD_DIR'
+      or will be set to 'opencbdc-tx/build' by default.
+    $ ./test.sh
+
+    - Run integration tests but do not run unit tests.  Do not measure
+      test coverage. The build directory is set by an environment variable
+      called 'BUILD_DIR' or will be set to 'opencbdc-tx/build' by default.
+    $ ./test.sh -nu -nc
+
+    - Run unit tests and integration tests and measure test coverage.  Set the
+      build directory to 'mybuild'.
+    $ ./test.sh -d mybuild
+
+    - Run only the ArchiverTest tests and measure test coverage.
+    $ ./test.sh --gtest_filter=ArchiverTest*
+
+    - Run only the ArchiverTest.client test.  Repeat it 4 times.
+      Don't measure test coverage.
+    $ ./test.sh --gtest_filter=ArchiverTest.client --gtest_repeat=4 -nc
+
+    - Run only unit tests and don't measure test coverage.
+      Stop the tests if any test fails.
+    $ ./test.sh -ni -nc --gtest_break_on_failure
+	"
+}
+
+# Parse command-line arguments.
+RUN_UNIT_TESTS="true"
+RUN_INTEGRATION_TESTS="true"
+MEASURE_COVERAGE="true"
+GTEST_FLAGS=()
+while [[ $# -gt 0 ]]
+do
+    case $1 in
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        -d|--build-dir)
+            shift
+            ARG="$1"
+            if [[ $ARG == "" || ${ARG:0:1} == "-" ]]
+            then
+                echo -n "ERROR:  The -d flag was used, "
+                echo "but a valid build folder was not given."
+                exit 1
+            fi
+            BUILD_DIR=$ARG
+            shift
+            ;;
+        -ni|--no-integration-tests)
+            RUN_INTEGRATION_TESTS="false"
+            shift
+            ;;
+        -nu|--no-unit-tests)
+            RUN_UNIT_TESTS="false"
+            shift
+            ;;
+        -nc|--no-coverage)
+            MEASURE_COVERAGE="false"
+            shift
+            ;;
+        --gtest_*)
+            GTEST_FLAGS+=("$1")
+            shift
+            ;;
+        -*)
+            echo "ERROR:  unknown command-line option '${1}'."
+            exit 1
+            ;;
+        *)
```

Comment:
```suggestion
        *)
```

Is there a reason to separate these cases?

Reply:
Nope--no good reason.  Good catch!